Compile error with NVCC + OpenMPI

I'm seeing the following error in `request.h` when compiling a code with NVCC and OpenMPI:

`/home/choin/hoodoo/install/moose/libmesh/installed/include/timpi/request.h(112): error: expression must have a constant value`

It seems that the definition of `MPI_REQUEST_NULL` in OpenMPI is not considered compile-time constant by NVCC, which is:

```
OMPI_DECLSPEC extern struct ompi_predefined_request_t ompi_request_null;
#define OMPI_PREDEFINED_GLOBAL(type, global) (static_cast<type> (static_cast<void *> (&(global))))
#define MPI_REQUEST_NULL OMPI_PREDEFINED_GLOBAL(MPI_Request, ompi_request_null)
```

We might need a better way of defining `null_request` that is compatible across different MPI variants.

Thanks for a quick reply, but that leads to another failure:
`/home/choin/hoodoo/install/moose/libmesh/installed/include/timpi/request.h:112:24: error: ‘constexpr’ needed for in-class initialization of static data member ‘ompi_request_t* const TIMPI::Request::null_request’ of non-integral type [-fpermissive]
  112 |   static const request null_request = MPI_REQUEST_NULL;
      |                        ^~~~~~~~~~~~`

--- a/src/parallel/include/timpi/request.h
+++ b/src/parallel/include/timpi/request.h
@@ -108,11 +108,7 @@ public:
   // made from \p this have been cleaned up.
   void add_post_wait_work(PostWaitWork * work);
 
-#ifdef TIMPI_HAVE_MPI
-  static constexpr request null_request = MPI_REQUEST_NULL;
-#else
-  static constexpr request null_request = 0;
-#endif
+  static const request null_request;
 
 private:
   request _request;

--- a/src/parallel/src/request.C
+++ b/src/parallel/src/request.C
@@ -37,6 +37,12 @@
 namespace TIMPI
 {
 
+#ifdef TIMPI_HAVE_MPI
+const request Request::null_request = MPI_REQUEST_NULL;
+#else
+const request Request::null_request = 0;
+#endif
+
 // ------------------------------------------------------------
 // Request member functions
 Request::Request () :

